docs/integrations/react.rst

```diff
@@ -24,30 +24,69 @@ At this point, Raven is ready to capture any uncaught exception.
+are an important tool for defining the behavior of your application in the face of errors. Be sure to send errors they catch to 
+Sentry using ``Raven.captureException``, and optionally this is also a great opportunity to surface `User Feedback <https://docs.sentry.io/learn/user-feedback/>`_
 
 .. code-block:: javascript
 
-    function logException(ex, context) {
-      Raven.captureException(ex, {
-        extra: context
-      });
-      /*eslint no-console:0*/
-      window.console && console.error && console.error(ex);
-    }
-
-Now in your components (or anywhere else), you can fail gracefully:
-
-.. code-block:: javascript
+    class ExampleBoundary extends Component {
+        constructor(props) {
+            super(props);
+            this.state = { error: null };
+        }
+        
+        componentDidCatch(error, errorInfo) {
+            this.setState({ error });
+            Raven.captureException(error, { extra: errorInfo });
+        }
 
-    var Component = React.createClass({
         render() {
-            try {
-                // ..
-            } catch (ex) {
-                logException(ex);
+            if (this.state.error) {
+            return (
+                <div
+                className="snap"
+                onClick={() => Raven.lastEventId() && Raven.showReportDialog()}>
+                <img src={oops} />
+                    <p>We're sorry — something's gone wrong.</p>
+                    <p>Our team has been notified, but click here fill out a report.</p>
+                </div>
+            );
+            } else {
+                //when there's not an error, render children untouched
+                return this.props.children; 
             }
         }
-    });
+    }
+
+.. code-block:: javascript
+    <div>
+        <ExampleBoundary>
+            <h2>Sidebar</h2>
+            <Widget/>
+        </ExampleBoundary>
+        <p> This content won't unmount when Widget throws. </p>
+    </div>
+
+One important thing to note about the behavior of error boundaries in development mode is that React will the rethrow errors they catch.
+This will result in errors being reported twice to Sentry with the above setup, but this won't occur in your production build.
+
+Read more about error boundaries `in this blog post <https://blog.sentry.io/2017/09/28/react-16-error-boundaries>`_.
+
+Redux
+----------------
+If you use `Redux <https://github.com/reactjs/redux>`_ there are some useful community maintained middleware packages 
+for annotating error reports with useful information, such as store state and recent actions:
+
+- `captbaritone/raven-for-redux <https://github.com/captbaritone/raven-for-redux>`_
+- `ngokevin/redux-raven-middleware <https://github.com/ngokevin/redux-raven-middleware>`_
+
+Redux Sagas Middleware
+----------------
+If you're using `Redux Saga <https://github.com/redux-saga/redux-saga>`_ be
+aware that it does not bubble errors up to the browsers uncaught exception
```

Comment:
> browser's

\+ dangling `handler` word (not sure if it's rendered 1:1 or lines will be realigned on the page)